controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaDefaultAuthHandler.java

```diff
@@ -0,0 +1,139 @@
+import java.util.stream.Collectors;
+import lombok.Data;
+import org.jasypt.util.password.StrongPasswordEncryptor;
+
+public class PravegaDefaultAuthHandler implements PravegaAuthHandler {
+    private static final String DEFAULT_NAME = "Pravega-Default";
+    private final Map<String, PravegaACls> userMap;
+
+    public PravegaDefaultAuthHandler() {
+        userMap = new ConcurrentHashMap<>();
+    }
+
+    private void loadPasswdFile(String userPasswdFile) {
+        try (FileReader reader = new FileReader(userPasswdFile);
+             BufferedReader lineReader = new BufferedReader(reader)) {
+            String line;
+            while ( !Strings.isNullOrEmpty(line = lineReader.readLine())) {
+                if (line.startsWith("#")) {
+                    continue;
+                }
+                String[] userFields = line.split(":");
+                if (userFields.length >= 2) {
+                    String acls;
+                    if (userFields.length == 2) {
+                        acls = "";
+                    } else {
+                        acls = userFields[2];
+                    }
+                    userMap.put(userFields[0], new PravegaACls(userFields[1], getAcls(acls)));
+                }
+            }
+        } catch (IOException e) {
+            throw new CompletionException(e);
+        }
+    }
+
+    @Override
+    public String getHandlerName() {
+        return DEFAULT_NAME;
+    }
+
+    @Override
+    public boolean authenticate(Map<String, String> headers) {
+        String userName = headers.get("username");
+        String passwd = headers.get("password");
+        Preconditions.checkArgument(userName != null, "Username not found in header");
+        Preconditions.checkArgument(passwd != null, "Password not found in header");
+
+        StrongPasswordEncryptor encryptor = new StrongPasswordEncryptor();
+        return userMap.containsKey(userName) && encryptor.checkPassword(passwd, userMap.get(userName).encryptedPassword);
+    }
+
+    @Override
+    public PravegaAccessControlEnum authorize(String resource, Map<String, String> headers) {
+        String userName = headers.get("username");
+        if (Strings.isNullOrEmpty(userName) || !userMap.containsKey(userName)) {
+            throw new CompletionException(new PravegaAuthenticationException(userName));
+        }
+        return authorizeForUser(userMap.get(userName), resource);
+
+    }
+
+    @Override
+    public void initialize(Object serverConfig) {
+        loadPasswdFile(((GRPCServerConfig) serverConfig).getUserPasswdFile());
+    }
+
+    private PravegaAccessControlEnum authorizeForUser(PravegaACls pravegaACls, String resource) {
+        PravegaAccessControlEnum retVal = PravegaAccessControlEnum.NONE;
+
+        for (PravegaAcl acl : pravegaACls.acls) {
+            if (acl.resource.equals(resource)) {
+                return acl.acl;
+            } else if (resource.startsWith(acl.resource)) {
```

Comment:
If I follow the logic correctly, this is a bug.  Suppose my user has a `READ_UPDATE` on `myscope`.   The resource passed in is `myscope2/stream1`.     `myscope2/stream1` starts with `myscope`, so it will return `READ_UPDATE` even though the user doesn't have access to `myscope2`.

I believe scopes and streams cannot be renamed after creation.  Can you confirm?

Reply:
This is correct. Will fix this and update the PR.